pkg/multicluster/gateway_wrapper.go

```diff
@@ -173,7 +174,7 @@ func (g *GatewayWrapper) GetClusterGateways() []ClusterGateway {
 		}
 		clusterGateways = append(clusterGateways, ClusterGateway{
 			Gateway:     gw,
-			ClusterName: clusterName,
+			ClusterName: g.ClusterID,
```

Comment:
This one should not be changed. When a control plane has been used to generate the Gateway status the cluster Names are retrieved from the status. 
This will make all the cluster specific endpoints identical and break the OCM integration.

Reply:
discussing in https://github.com/Kuadrant/kuadrant-operator/pull/493#discussion_r1537354557